src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaResourceDetector.cs

```diff
@@ -14,14 +14,26 @@ internal sealed class AWSLambdaResourceDetector : IResourceDetector
     /// <returns>Detected resource.</returns>
     public Resource Detect()
     {
-        var resourceAttributes = new List<KeyValuePair<string, object>>
+        var resourceAttributes = new List<KeyValuePair<string, object>>(4)
         {
             new(AWSLambdaSemanticConventions.AttributeCloudProvider, AWSLambdaUtils.GetCloudProvider()),
-            new(AWSLambdaSemanticConventions.AttributeCloudRegion, AWSLambdaUtils.GetAWSRegion()),
-            new(AWSLambdaSemanticConventions.AttributeFaasName, AWSLambdaUtils.GetFunctionName()),
-            new(AWSLambdaSemanticConventions.AttributeFaasVersion, AWSLambdaUtils.GetFunctionVersion()),
         };
 
+        if (AWSLambdaUtils.GetAWSRegion() is { } region)
```

Comment:
This looks arcane, would this work too?

```suggestion
        if (AWSLambdaUtils.GetAWSRegion() is not null region)
```

Reply:
Yes it would, they mean the same thing. I'll change them.

Reply:
Actually, no, that isn't valid C#.

Options are:

1. What was already there;
2. The commit I just pushed to do `{ Length: > 0 }` instead;
3. Explicitly declare the variables and check them (more verbose IMHO).

```csharp
var region = AWSLambdaUtils.GetAWSRegion();
if (region is not null)
{
    resourceAttributes.Add(new(AWSLambdaSemanticConventions.AttributeCloudRegion, region));
}

// etc
```

Reply:
Sorry, if `is not null` does not work, I'm totally fine with the `{ }`, it's probably an idiom people who write more modern C# than me are familiar with 😃

Reply:
OK, in that case I'll change it back as it's terser.